src/components/molecules/OakModalCenter/OakModalCenter.tsx

```diff
@@ -0,0 +1,257 @@
+import { parseSpacing } from "@/styles/helpers/parseSpacing";
+
+export type OakModalCenterProps = {
+  /**
+   * The content of the modal. Use with `<OakModalCenterBody>` for best results.
+   */
+  children: ReactNode;
+  /**
+   * Determines whether to show the modal or not
+   */
+  isOpen: boolean;
+  /**
+   * Override HTMLAttributes & OakFlex props for the modal container
+   */
+  modalFlexProps?: Partial<OakFlexProps & HTMLAttributes<Element>>;
+  /**
+   * Override HTMLAttributes & OakFlex props for the backdrop container
+   */
+  backdropFlexProps?: Partial<OakFlexProps & HTMLAttributes<Element>>;
+  /**
+   * If true, clicking the backdrop will not call onClose
+   */
+  disableBackdropClick?: boolean;
+  /**
+   * If true, pressing the escape key will not call onClose
+   */
+  disableEscapeKey?: boolean;
+  /**
+   * If true, the close button will be hidden
+   */
+  hideCloseButton?: boolean;
+  /**
+   * Called when the modal is closed via the close button, backdrop click, or escape key
+   */
+  onClose: () => void;
+  /**
+   * The DOM container to render the modal portal into.
+   */
+  domContainer?: Element;
+  /**
+   * Slot for the footer of the modal
+   */
+  footerSlot?: ReactNode;
+};
+
+const FocusOnBox = styled(FocusOn)`
+  ${oakBoxCss}
+`;
+
+const FadeInFlex = styled(OakFlex)<{ $state: TransitionStatus }>`
+  opacity: ${({ $state }) => {
+    switch ($state) {
+      case "entered":
+      case "entering":
+        return "1";
+      default:
+        return "0";
+    }
+  }};
+  transition: opacity 0s;
+`;
+
+/**
+ * Centered modal dialog with trapped focus, close button, backdrop click, and escape key handling.
+ *
+ * ## Props
+ *
+ * - **isOpen** \-                  If true the modal will be open, if false it will be closed
+ * - **onClose** \-                 Called when the modal is closed via the close button, backdrop click, or escape key
+ * - **children** \-                The content of the modal. Use with `<OakModalCenterBody>` for best results.
+ * - **domContainer** \-            The DOM container to render the modal portal into.
+ * - **disableBackdropClick?** \-   If true, clicking the backdrop will not call onClose
+ * - **disableEscapeKey?** \-       If true, pressing the escape key will not call onClose
+ * - **hideCloseButton?** \-        If true, the close button will be hidden
+ * - **modalFlexProps?** \-         Override HTMLAttributes & OakFlex props for the modal container
+ * - **backdropFlexProps?** \-      Override HTMLAttributes & OakFlex props for the backdrop container
+ * - **footerSlot?** \-             Fixed area at the bottom of the modal, this will remain fixed in view if the content is scrollable
+ */
+export const OakModalCenter = ({
+  children,
+  domContainer = document.body,
+  isOpen,
+  onClose,
+  disableBackdropClick = false,
+  disableEscapeKey = false,
+  hideCloseButton = false,
+  modalFlexProps,
+  backdropFlexProps,
```

Comment:
Not sure if this is a pattern that we have used before, but these allow the consumer to extend or override the modal and backdrop properties. For adjusting styles or aria labels etc.